### PR TITLE
create retriever factory

### PR DESCRIPTION
**Pull request by debermudez**

### CODE REVIEW

File: genai-perf/genai_perf/inputs/inputs.py [hunk 1]
```diff
@@ -59,360 +48,28 @@ def create_inputs(self) -> Dict:
...
-        with open(self.config.input_filename, mode="r", newline=None) as file:
-            for line in file:
-                if line.strip():
-                    # None if not provided
-                    prompt = load_json_str(line).get("text_input")
-                    image = load_json_str(line).get("image")
-                    prompts.append(prompt.strip() if prompt else prompt)
-                    images.append(image.strip() if image else image)
-        return prompts, images
+        self._check_for_dataset_name_if_input_type_is_url()
```
Comment (dyastremsky) [inline genai-perf/genai_perf/inputs/inputs.py:62]: Why was the try/catch block removed?

File: genai-perf/tests/test_input_embeddings.py [hunk 2]
```diff
@@ -55,9 +59,9 @@ def test_get_input_dataset_from_embeddings_file(self, mock_file, mock_exists):
             batch_size=batch_size,
             num_prompts=100,
         )
-        inputs = Inputs(config)
+        factory = InputRetrieverFactory(config)
```
Comment (dyastremsky) [inline genai-perf/tests/test_input_embeddings.py:62]: Could you use a different variable name? Factory is too generic.

File: genai-perf/tests/test_input_rankings.py [hunk 3]
```diff
@@ -76,8 +77,8 @@ def test_get_input_dataset_from_rankings_file(self, mock_file):
             batch_size=batch_size,
             num_prompts=100,
         )
-        inputs = Inputs(config)
-        dataset = inputs._get_input_dataset_from_rankings_files(
+        irf = InputRetrieverFactory(config)
```
Comment (dyastremsky) [inline genai-perf/tests/test_input_rankings.py:80]: Could you use a different variable name and keep it consistent with the other test file? Irf is unintuitive, especially if you see it later in the code.

**Review by dyastremsky — COMMENTED** [review]
Fantastic work! A few small comments.

File: genai-perf/tests/test_input_embeddings.py (hunk 2 above)
Comment (debermudez) [reply genai-perf/tests/test_input_embeddings.py:62]: I have often seen short tests use generic names.
Ill update this and avoid that practice in the future.

Updated to `input_retriever_factory`

File: genai-perf/tests/test_input_rankings.py (hunk 3 above)
Comment (debermudez) [reply genai-perf/tests/test_input_rankings.py]: Updated to `input_retriever_factory`

File: genai-perf/tests/test_input_embeddings.py (hunk 2 above)
Comment (dyastremsky) [reply genai-perf/tests/test_input_embeddings.py]: Thanks! Could you do this for all of the tests? I think test_input_retriever_factory is the last one.

I think people used to prioritize conciseness more, but readability is always important, especially if you might use the variable more than 1-2 lines below its declaration. However, with code completion tools now widely available, the emphasis is shifting (at least in my opinion) to be readability >>> conciseness.

Comment (debermudez) [reply genai-perf/tests/test_input_embeddings.py]: done.

I agree with you. 
I didn't think this was really unreadable but if you are saying it was harder to follow, that is justification enough to change. Thanks for calling it out.

**Review by dyastremsky — APPROVED** [review]
Excellent work!

**Pull request merged**